builder/dockerfile/internals.go

```diff
@@ -624,7 +624,16 @@ func (b *Builder) readDockerfile() error {
 	// Note that this assumes the Dockerfile has been read into memory and
 	// is now safe to be removed.
 	if dockerIgnore, ok := b.context.(builder.DockerIgnoreContext); ok {
-		dockerIgnore.Process([]string{b.options.Dockerfile})
+		ignoreErr := false
+		ignoreFile := b.options.Ignorefile
+		if ignoreFile == "" {
+			ignoreFile = builder.DefaultDockerignoreFilename
+			ignoreErr = true
+		}
+		err = dockerIgnore.Process(ignoreFile, []string{b.options.Dockerfile})
+		if !ignoreErr && err != nil {
+			return err
+		}
```

Comment:
Is there any reason this "ignoreErr" logic isn't in Process()?  I worry about putting it here because it assumes that the only possible error from Process() is that .dockerignore doesn't exist - which right now is true. But, if at some point in the future another error is returned then the next person to touch this code **might** do something really hacky like start checking for certain errors - or worse, forget to even check at all and introduce a bug.

Its not a big thing, but I would prefer if all of this new logic were moved into Process() so that we can add the ignoreErr logic right at the `os.IsNotExist(err)` line of code and avoid this potential headache later on.  I think the code will be smaller/easier too.